pkg/kubelet/container/runtime_cache_test.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
```

Comment:
> Most people understand how a simple cache works even without tests :)

I beg to differ. It's easy to make mistakes while touching code like this ;)
The tests at least provide *some* sanity check for this kind of purposes.

I understand the value of demonstrating and verifying the pod worker use case, but I also found the test to be confusing and clunky as I pointed out earlier. The current implementation updatingThread'd make the test unnecessarily complicated. Refactoring runtimeCache may help with the the testability, but I am hesitant to prioritize that...